src/binding/db_registry.cpp

```diff
@@ -208,6 +208,45 @@ void DBRegistry::PurgeAll() {
 	}
 }
 
+/**
+ * Shutdown will force all databases to flush in-memory data to disk and purge the registry.
+ */
+void DBRegistry::Shutdown() {
+	if (instance) {
+		std::vector<std::shared_ptr<DBDescriptor>> descriptorsToClose;
+
+		{
+			std::lock_guard<std::mutex> lock(instance->databasesMutex);
+			DEBUG_LOG("%p DBRegistry::Shutdown Shutting down %zu databases\n", instance.get(), instance->databases.size())
+
+			// Collect all descriptors to close
+			for (auto& [path, entry] : instance->databases) {
+				if (entry.descriptor) {
+					descriptorsToClose.push_back(entry.descriptor);
+				}
+			}
+		}
+
+		// Close all descriptors without holding the lock
+		for (auto& descriptor : descriptorsToClose) {
+			DEBUG_LOG("%p DBRegistry::Shutdown Closing database: %s\n", instance.get(), descriptor->path.c_str())
+			descriptor->closing.store(true);
```

Comment:
I'm inclined to add a `close()` method so that we don't leak these internal things such as `closing` and `db`. I'm surprised we don't already have a `close()` method. I probably did have one at some point.

What about closing all of the `DBDescriptor`'s active `DBHandles` and closing the transaction log stores?

Reply:
Do you want to do that and handle the `CloseDB()` path?